todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsFragment.java

```diff
@@ -59,24 +68,68 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onResume() {
         super.onResume();
-        mPresenter.subscribe();
+        bind();
     }
 
     @Override
     public void onPause() {
         super.onPause();
-        mPresenter.unsubscribe();
+        unbind();
     }
 
-    @Override
-    public void setProgressIndicator(boolean active) {
+    private void bind() {
+        mSubscription = new CompositeSubscription();
+
+        mSubscription.add(mViewModel.getProgressIndicator()
+                .subscribe(new Subscriber<Boolean>() {
```

Comment:
We could make this more functional by defining three `Action` instances for `onCompleted()`, `onError()` and `onNext()`.